tests/drivers/build_all/sensor/uart.dtsi

```diff
@@ -15,3 +16,9 @@ test_uart_mhz19b {
 test_uart_pms7003 {
 	compatible = "plantower,pms7003";
 };
+
+test_uart_grow_r502a: grow_r502a@ffffffff {
```

Comment:
This shouldn't have an address

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/40394#discussion_r943217396  there is a previous suggestion to use reg property from `uart-controller.yaml` , looking at the yaml reg property comes from `base.yaml` there reg property is optional, so either way, using `uart-device.yaml `and uart-controller.yaml makes no significant difference. but the actual issue is, this fingerprint sensor has a 4byte address byte going into data package. the suggestion was to remove custom defined addr property in `hzgrow,r502a.yaml ` and to use reg property , I make the changes according to that suggestion.

Here, coming to this issue, I need to use #address-cells = <1>; and optional #size-cells = <0>; property inside uart parent node to make the sample compilable. looking at how to not include #*-cells properties inside uart parent node is like below:

```
test_uart_grow_r502a {
        #address-cells=<1>;
        #size-cells=<0>;
        grow_r502a@0 {
        	compatible = "hzgrow,grow-r502a";
        	label = "GROW_R502A";
        	reg=<0x0>;
        	int-gpios = <&test_gpio 0 0>;
        };
};
```
is it okay to use node like the above snippet? @MaureenHelm